preprocessing/structs.ts

```diff
@@ -96,121 +161,159 @@ const id_field = (name: string) => {
+      */
+      { name: "gasmax", bits: 24, type: "uint" },
+      /* * `dead`: if necessary, Mangrove can be entirely deactivated by governance (offers can still be withdrawn and bounties can still be withdrawn). Once killed, Mangrove must be redeployed. It cannot be resurrected. */
+      { name: "dead", bits: 1, type: "bool" },
+      /* * `maxRecursionDepth` is the maximum number of times a market order can recursively execute offers. This is a protection against stack overflows. */
+      { name: "maxRecursionDepth", bits: 8, type: "uint" },      
+      /* * `maxGasreqForFailingOffers` is the maximum gasreq failing offers can consume in total. This is used in a protection against failing offers collectively consuming the block gas limit in a market order. Setting it too high would make it possible for successive failing offers to consume up to that limit then trigger a revert (thus the failing offer would not be removed). During a market order, Mangrove keeps a running sum of the `gasreq` of the failing offers it has executed, and stops the market order when that sum exceeds `maxGasreqForFailingOffers`. */
+      { name: "maxGasreqForFailingOffers", bits: 32, type: "uint" },      
+    ],
+  },
+
+  /* ## Local configuration */
+  local: {
+    fields: [
+      /* * An offer list is not `active` by default, but may be activated/deactivated by governance. */
+      { name: "active", bits: 1, type: "bool" },
+      /* * `fee`, in basis points, of `outbound_tkn` given to the taker. This fee is sent to Mangrove. Fee is capped to ~2.5%. */
+      { name: "fee", bits: 8, type: "uint" },
+      /* * `density` is similar to a 'dust' parameter. We prevent spamming of low-volume offers by asking for a minimum 'density' in `outbound_tkn` per gas requested. For instance, if `density` is worth 10, `offer_gasbase == 5000`, an offer with `gasreq == 30000` must promise at least _10 × (30000 + 5000) = 350000_ `outbound_tkn`. _9 bits wide_.
+
+      We store the density as a float with 2 bits for the mantissa, 7 for the exponent, and an exponent bias of 32, so that density ranges from $2^{-32}$ to $1.75 \times 2^{95}$. For more information, see `DensityLib`.
+      
+      */
+      { name: "density", bits: 9, type: "Density", underlyingType: "uint"},
+      /* To save on gas, Mangrove cache the entire tick tree branch of the bin that contains the best offer in each offer list's `local` parameter. Taken together, `binPosInLeaf`, `level3`, `level2`, `level1` and `root` provide the following info:
+      - What the current bin is (see `BinLib.bestBinFromLocal`)
+      - When a leaf is emptied and the next offer must be fetched, the information in the fields `level3`, `level2`, `level1` and `root` avoid multiple storage reads
```

Comment:
Spacing is weird here in the generated docs:

![billede](https://github.com/mangrovedao/mangrove-core/assets/3458856/f9ade2ca-437d-46cb-b7c0-b80c21ea8e6b)

Reply:
Ah, just realized that the first two bullets pertain to a different set of fields than the third.
Not that important I guess.